ci/poll-extension-availability.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Query the installed extension until the expected version is available.
+# Inspired from: https://github.com/microsoft/azure-devops-extension-tasks/issues/189#issue-820340124
+
+set -e
+
+SERVICE_URL=
+SERVICE_TOKEN=
+TASK_ID=
+TASK_VERSION=
+
+while getopts t:s:c:v: flag; do
+    case "${flag}" in
+        s) SERVICE_URL="${OPTARG}";;
+        t) SERVICE_TOKEN="${OPTARG}";;
+        c) TASK_ID="${OPTARG}";;
+        v) TASK_VERSION="${OPTARG}";;
+    esac
+done
+
+attempts=0
+max=25
+
+until [ $attempts -gt $max ];
+do
+    extensions=$(tfx build tasks list --authType pat --service-url $SERVICE_URL -t $SERVICE_TOKEN --no-color --json)
+    installed_version=$(echo $extensions | jq -r ".[] | select(.contributionIdentifier == \"$TASK_ID\") | .contributionVersion")
```

Comment:
I'm not sure to understand these commands. Could we add a few comment lines about them?